### PR TITLE
LR Scheduler implemented

### LINKED ISSUE #135: Implement configurable learning rate scheduler
Learning Rate scheduler ([described here](https://pytorch.org/docs/stable/optim.html#how-to-adjust-learning-rate)) to be defined in the config.

Also note that learning rate schedulers can be "stacked" as in the second example code in the link listed above. This increases the complexity of the problem both in terms of the way we define things in the config file and how we implement the scheduler in the code. 

@aritraghsh09 has already mentioned that indeed people do tend to use LR schedulers in the real world, but, IIRC, he hasn't yet used them. 

Finally - if we do implement configurable LR schedulers, we should add the LR value to the tensorboard output.

### PR DESCRIPTION
**Pull request by SamSandwich07**
Allows users to configure a single basic (i.e. not Chained or Sequential) learning rate scheduler in the config. Closes #135.

### CODE REVIEW

File: src/hyrax/pytorch_ignite.py [hunk 1]
```diff
@@ -729,6 +733,17 @@ def log_epoch_metrics(trainer):
                 )
                 mlflow.log_metrics({f"training/epoch/{m}": epoch_metrics[m]}, step=epoch_number)
 
+    @trainer.on(HyraxEvents.HYRAX_EPOCH_COMPLETED)
+    def scheduler_step(trainer):
+        if model.scheduler:
+            if not hasattr(model, "lrs"):
+                model.lrs = []
```
[the remark this answers is not in the document]
Comment (drewoldag) [reply src/hyrax/pytorch_ignite.py]: Oh, ok, I see this being used in the unit tests. Yeah, I would again advocate for this to be a "private" variable, and named in a way that it would be extremely unlikely to conflict with a existing variable.

**Review by drewoldag — COMMENTED** [review]
So far this is looking good. Left a couple of comments.

File: src/hyrax/verbs/train.py [hunk 2]
```diff
@@ -150,3 +150,9 @@ def _log_params(config, results_dir):
         mlflow.log_param("optimizer", optimizer_name)
         if optimizer_name in config:
             mlflow.log_params(config[optimizer_name])
+
+        # Log the scheduler params
+        scheduler_name = config["scheduler"]["name"]
```
[the remark this answers is not in the document]
Comment (drewoldag) [reply src/hyrax/verbs/train.py]: We chatted about this. And @SamSandwich07 created this issue to track the future work: https://github.com/lincc-frameworks/hyrax/issues/685

**Review by drewoldag — APPROVED** [review]
This is looks good to me. Thanks for putting this together!
